### PR TITLE
Renamed Argument Labels

### PR DESCRIPTION
**Pull request by TaeJoongYoon**
I edit functions in Router name and arguments according to API Design Guidelines

```swift

// From
routingActionsForTransitionFrom(_ oldRoute:,newRoute:)
// To
routingActionsForTransition(from oldRoute:,to newRoute:)

// From
routableIndexForRouteSegment(_ segment:)
// To
routableIndex(for segment:)
```

### CODE REVIEW

**Comment by mjarvis** [comment]
@TaeJoongYoon Thanks for contributing!

When we rename or remove any public facing interfaces, we need to provide deprecated forwards from the old functions to avoid breaking backwards compatibility.

For example:
```
@available(*, deprecated, message: "Deprecated in favor of `routableIndex(for:)`")
static func routableIndexForRouteSegment(_ segment: Int) -> Int {
        return routableIndex(for: segment)
}
```

**Comment by DivineDominion** [comment]
@TaeJoongYoon Travis reports `Use of unresolved identifier 'routableIndexForRouteSegment'` -- have you overlooked an occurence during the renaming process?

**Comment by TaeJoongYoon** [comment]
@mjarvis
It's okay! :) If needed, I will add it.

@DivineDominion
I'm sorry. I missed it one line. I edited it and commit it!

**Comment by DivineDominion** [comment]
@TaeJoongYoon There seems to be more typos :) 
https://travis-ci.org/ReSwift/ReSwift-Router/jobs/535598154#L2080

The `for:` is missing. (I suggest you ⌘U run tests locally too to catch this)

**Comment by mjarvis** [comment]
It would also be beneficial to add an entry to `Changelog.md` referencing these breaking changes

**Comment by TaeJoongYoon** [comment]
@DivineDominion I'm sorry.. test succeed.

@mjarvis Do I add it?

**Comment by DivineDominion** [comment]
@TaeJoongYoon Great!

Regarding the CHANGELOG: yes, please! For example:

```md
- Renamed argument labels for modern Swift conformance (#116) - @TaeJoongYoon
```

Or whatever you find fitting.

**Comment by TaeJoongYoon** [comment]
@DivineDominion I added it with comments!

File: Changelog.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
     - Removes the compatibility of this with `ReSwift-Recorder`, which itself is being deprecated.
     - Ensures compatibility with latest versions of `ReSwift` which have removed these types.
+- Renamed argument labels for modern Swift conformance ([#116](<https://github.com/ReSwift/ReSwift-Router/pull/115>)) - @TaeJoongYoon
```
Comment (mjarvis) [inline Changelog.md]:
```suggestion
- Renamed argument labels for modern Swift conformance ([#116](<https://github.com/ReSwift/ReSwift-Router/pull/116>)) - @TaeJoongYoon
```

(fixed the pull request number in the URL)

Comment (TaeJoongYoon) [reply Changelog.md]: My apology for the typo...

Comment (TaeJoongYoon) [reply Changelog.md]: Do I have to do anything extra?

Comment (DivineDominion) [reply Changelog.md]: I don't see anything else. I'll merge it once this is fixed 👍

Comment (TaeJoongYoon) [reply Changelog.md]: Okay. I fixed it :)

**Pull request merged**